lc-0000-template/src/test/java/solution/CustomCsvArgTest.java

```diff
@@ -23,14 +53,27 @@ static class MyArg {
         public final int[] values;
     }
 
+    // Template to test 1D array inputs/outputs:
     @ParameterizedTest
     @CsvSource(value = {
             "[1,2,3]; [10,10,10]; [11,12,13]",
-            "[1,2,3]; [19,18,17]; [20,20,20]",
+            "[1,2,3]; [19,18,17]; [20,20,20]"
     }, delimiter = ';')
     void testWithMyArgs(MyArg a, MyArg b, MyArg c) {
         Solution s = new Solution();
         assertArrayEquals(c.values, s.addAll(a.values, b.values));
     }
 
+    // Template to test 2D array inputs/outputs:
+
+    @ParameterizedTest
+    @CsvSource(value = {
+            "[] []; 0",
+            "[1,2,3] [4,5,6] [7,8,9]; 45"
```

Comment:
I would like there to be a more visual way to represent 2D arrays if possible. This is a want not need so if the implementation is too complex then it's not worth it. Something like:

```
    @CsvSource(value = {
            "[]
             [];
             0",
            "[1,2,3]
             [4,5,6]
             [7,8,9];
             45"
    }, delimiter = ';')
```

This would be especially useful for matrix problems such as the ones that deal with modifying an image. Having side by side arrays would make it easier to work with visually.

Reply:
We agreed to either pass the int matrix directly into the test, or use something like this: https://stackoverflow.com/a/878704